src/formatters/default_formatter.erl

```diff
@@ -1342,6 +1352,51 @@ lay_application(Name, Arguments, SpacesWithinParentheses, Ctxt) ->
             maybe_parentheses(D, Prec, Ctxt)
     end.
 
+%% @doc Might produce a new AST on which the functions in the export list
+%%      are sorted depending on what 'sort_function_exports' was set to:
+%%          - alphabetically, if set to 'alphabetically'
+%%          - left as it is, if set to 'false'
+sort_function_exports([Arguments0], SortFun) ->
+    Attrs = erl_syntax:get_pos(Arguments0),
+    case erl_anno:text(Attrs) of
+        %% If the attribute is indeed a export list, and the rule was enabled,
+        %% we sort the functions; otherwise, we ignore the attribute.
+        "export" ->
```

Comment:
There must be a better way to do this than using text.
Can't we use something like [`erl_syntax:attribute_name/1`](https://www.erlang.org/doc/man/erl_syntax.html#attribute_name-1)?

Reply:
Nope, sadly we cannot, AFAIK.

`erl_syntax:attribute_name/1` works for `attribute` AST nodes, and the `Arguments0` in this context, is the list of argument subtrees of the attribute node, [obtained by doing](https://github.com/AdRoll/rebar3_format/blob/42aae42b385f0a23161140b82e52f698e9130ac0/src/formatters/default_formatter.erl#L416)
```erlang
case {Tag, erl_syntax:attribute_arguments(Node)} of
  {export, Args} ->
    ...
    %% here we call default_formatter:lay_application/3 with the Args, but not the 'export' tag
```

This list of arguments looks like this, for that clause of the `case ... of`, but `default_formatter:call lay_application/3` is called with many other argument values, so we have to re-check again that we have been indeed given the arguments related to a export list, and that information can be verified by making use of the `#attr{}` record either from the subtree itself, or from each of the subtrees that represent the export items. E.g.:
```erlang
#tree{type = list,
      attr = #attr{pos = [{text,"export"},{location,4}],
                   ann = [],com = none},
      data = #list{prefix = [#tree{type = arity_qualifier,
                                   attr = #attr{pos = [{text,"export"},{location,4}],
                                                ann = [],com = none},
                                   data = #arity_qualifier{body = #tree{type = atom,
                                                                        attr = #attr{pos = 0,ann = [],com = none},
                                                                        data = not_found},
                                                           arity = #tree{type = integer,
                                                                         attr = #attr{pos = 0,ann = [],com = none},
                                                                         data = 1}}},
                             #tree{type = arity_qualifier,
                                   attr = #attr{pos = [{text,"export"},{location,4}],
                                                ann = [],com = none},
                                   data = #arity_qualifier{body = #tree{type = atom,
                                                                        attr = #attr{pos = 0,ann = [],com = none},
                                                                        data = no_updates},
                                                           arity = #tree{type = integer,
                                                                         attr = #attr{pos = 0,ann = [],com = none},
                                                                         data = 1}}},
                             ...],
                   suffix = none}}
```

Reply:
OOOOH! So we only get the list of arity qualifiers here.
We generally solve this kind of problem by passing down something in the context.
It's like what we do with `force_indentation = true`.
You can do something like that:
1. add a new field to `#ctxt`: `use_sorting = false :: boolean()`.
2. on `lay_no_comments/2`, when you figure out you're laying an export list, set that field to `true` (just like we do with `force_indentation`).
3. Check that field on `lay_application` instead of checking the text of the list of nodes.

That has the added benefit of allowing us to sort other lists of qualifiers (like `-dialyzer` or `-export_type` ;)).

As a matter of fact, a different approach would be to rename your config as `sort_qualifier_lists`… and then… if you get a list full of `#arity_qualifier`s… sort it out, regardless of its origin. 🤔

Reply:
Gotcha, that does make sense. But from what I understand, wouldn't that turn the config on by default, instead of having it disabled and it being optional? I worked on the PR on the basis that no config/rule is used by default when added to the formatter.

Work for me either way, and I love the idea of sorting other `#arity_qualifier{}`s :P

Reply:
No, no… It's not the same thing… Do not turn the flag (`sort_function_exports`) on by default. Just turn a flag that says "we should **honour** `sort_function_exports`" when it's an `export`. Keep the "should check" flag `false` by default and only turn it on when calling `lay_application` after determining that we're talking about an `export`.

Reply:
And I think that the effort of changing this to a config flag that sorts all arity qualifier lists is much better, anyway.

Reply:
If it is okay with you, I will tackle the "sort all arity qualifiers" on a future PR. I would rather work iteratively on the formatter and get myself used to its codebase, than once in a blue moon :)

Reply:
It's not ok for me, sorry. I wouldn't want to add an option (`sort_function_exports`) in a version and remove it/replace it in the next one with a more comprehensive one (`sort_arity_qualifiers`).
So… a different PR is fine, but it has to happen _before_ the next release.